From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect vhost_log_write
Patch-mainline: never, kabi
References: kabi

In networking-stable-19_01_26, commit
cc5e710759470bc7f3c61d11fd54586f15fdbdf4 (vhost: log dirty page
correctly) added two parameters to vhost_log_write.  This indeed
changed the checksum of this exported function and the kABI checker
now complains.

As this is only a vhost internal function, just hide the change from
the kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/vhost/vhost.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -210,8 +210,12 @@ bool vhost_vq_avail_empty(struct vhost_d
 bool vhost_enable_notify(struct vhost_dev *, struct vhost_virtqueue *);
 
 int vhost_log_write(struct vhost_virtqueue *vq, struct vhost_log *log,
+#ifdef __GENKSYMS__
+		    unsigned int log_num, u64 len);
+#else
 		    unsigned int log_num, u64 len,
 		    struct iovec *iov, int count);
+#endif
 int vq_iotlb_prefetch(struct vhost_virtqueue *vq);
 
 struct vhost_msg_node *vhost_new_msg(struct vhost_virtqueue *vq, int type);
